### PR TITLE
Initially add conversion docs and updates for search contexts

### PR DESCRIPTION
**Pull request by quinnkeast**
Initial draft of docs for introducing search contexts for Enterprise.

- A bit of reworking of how search contexts are explained (to bridge Cloud and Enterprise differences right now)
- Clarification that version contexts are sunsetting
- Instructions for converting existing version contexts to search contexts

I believe we should also add a section on creating and using search contexts, now that that's an option for users when search contexts are enabled. I plan to write this as well and include in the same PR.

### CODE REVIEW

File: doc/code_search/explanations/search_contexts.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Sourcegraph Cloud
+
+[Sourcegraph Cloud](https://sourcegraph.com/search) lets you search inside your public code on GitHub or GitLab, and inside any open-source project on GitHub.
+
+Note that you can search across a maximum of 50 repositories at once using Sourcegraph Cloud. To search across more than 50 repositories at once or to search your organization's internal code, [run your own Sourcegraph instance](../../../admin/install/index.md).
+
+## Search contexts <span class="badge badge-primary">experimental</span>
```
Comment (novoselrok) [inline doc/code_search/explanations/search_contexts.md:7]: Should this be `beta` as well?

File: doc/code_search/how-to/search_contexts.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        "repository": "github.com/sourcegraph/sourcegraph",
+        "revisions": [
+          "3.15"
+        ]
+      }, {
+        "repository": "github.com/sourcegraph/src-cli",
+        "revisions": [
+          "3.11.2"
+        ]
+      },
```
Comment (novoselrok) [inline doc/code_search/how-to/search_contexts.md:59]: Remove the trailing comma (it produces an error unfortunately).

File: doc/code_search/how-to/search_contexts.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+- In the **Owner** field, choose whether you will own the context or if it will be global to the Sourcegraph instance. **Note**: At present, the owner of a search context cannot be changed after being created.
+- In the **Context name** field, type in a short, semantic name for the context. The name can be 32 characters max, and contain alphanumeric and `.` `_` `/` `-` characters.
+- Optionally, enter a **Description** for the context. Markdown is supported.
+- Choose the **Visibility** of this context.
+  - Public contexts are available to everyone on the Sourcegraph instance. Note that private repositories will only be visible to users that have permission to view the repository via the code host.
+  - Private contexts can only be viewed by their owner, or in the case being globally owned, by site admins.
+- In the **Repositories and revisions** configuration, define which repositories and revisions should be included in the search context. Press **Add repository** to quickly add a template to the configuration.
+  - Define repositories with valid URLs.
+  - Define revisions as objects in an array. To specify the default branch, you can set `"rev"` to `"HEAD"` or `""`.
```
Comment (novoselrok) [inline doc/code_search/how-to/search_contexts.md:43]: Define revisions as strings in an array. To specify a default branch use `HEAD`.

File: doc/code_search/how-to/search_contexts.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# Search contexts
+
+Search contexts help you search the code you care about on Sourcegraph. A search context represents a set of repositories at specific revisions on a Sourcegraph instance that will be targeted by search queries by default.
+
+Every search on Sourcegraph uses a search context. Search contexts can be defined with the contexts selector shown in the search input, or entered directly in a search query.
```
Comment (novoselrok) [inline doc/code_search/how-to/search_contexts.md:5]: See my comment above.

File: doc/code_search/explanations/features.md [hunk 5]
```diff
@@ -123,21 +118,32 @@ Indexing multiple branches will add additional resource requirements to Sourcegr
 
 > NOTE: All revisions specified in version contexts are also indexed.
 
-## Search contexts <span class="badge badge-primary">experimental</span>
+## Search contexts <span class="badge badge-primary">beta</span>
```
[the remark this answers is not in the document]
Comment (quinnkeast) [reply doc/code_search/explanations/features.md]: Swapped to "experimental" everywhere to push this back for now.

File: doc/code_search/explanations/features.md [hunk 6]
```diff
@@ -125,19 +120,34 @@ Indexing multiple branches will add additional resource requirements to Sourcegr
...
-Sourcegraph Cloud currently supports two search contexts: 
+**Sourcegraph Cloud** currently supports two pre-set search contexts: 
 
-- Your personal context, `context:@username`, which automatically includes all repositories you add to the Sourcegraph instance.
+- Your personal context, `context:@username`, which automatically includes all repositories you add to Sourcegraph Cloud.
+- The global context, `context:global`, which includes all repositories on Sourcegraph Cloud.
+
+If no search context is specified, `context:global` is used.
+
+**Private Sourcegraph instances** support custom search contexts:
```
Comment (lguychard) [inline doc/code_search/explanations/features.md:134]: Doesn't Cloud also support custom contexts?

**Review by lguychard — APPROVED** [review]
LGTM! 👍

File: doc/code_search/explanations/features.md (hunk 6 above)
Comment (novoselrok) [reply doc/code_search/explanations/features.md:134]: Correct, it is not enabled by default. If the user enables it manually on Cloud, then they can create custom contexts.

**Pull request merged**